lib/evss/disability_compensation_form/form4142_processor.rb

```diff
@@ -20,8 +20,40 @@ def initialize(submission, jid)
 
       protected
 
+      def transform_provider_facilities(incoming_data)
+        provider_facilities = incoming_data['providerFacility']
+        return incoming_data if provider_facilities.blank?
+
+        incoming_data['providerFacility'] = provider_facilities.map do |facility|
+          treated_disability_hash = facility['treatedDisabilityNames']
+          if treated_disability_hash.blank?
+            facility.merge('conditionsTreated' => '')
+          else
+            treated_conditions = treated_disability_hash.select { |_, checked| checked }.keys.join(', ')
+            facility.merge('conditionsTreated' => treated_conditions)
+          end
```

Comment:
The following avoids some of the `facility.merge('conditionsTreated' => ` duplication:

```suggestion
          treated_conditions = if treated_disability_hash.blank?
                           ''
                         else
                           treated_disability_hash.select { |_, checked| checked }.keys.join(', ')
                         end
    facility.merge('conditionsTreated' => treated_conditions)
```